kyuubi-server/src/test/scala/org/apache/kyuubi/parser/trino/KyuubiTrinoFeParserSuite.scala

```diff
@@ -108,4 +108,115 @@ class KyuubiTrinoFeParserSuite extends KyuubiFunSuite {
+        | SELECT TABLE_CAT, TABLE_SCHEM, TABLE_NAME, TABLE_TYPE, REMARKS,
+        | TYPE_CAT, TYPE_SCHEM, TYPE_NAME,
+        | SELF_REFERENCING_COL_NAME, REF_GENERATION
+        | FROM system.jdbc.tables
+        | WHERE TABLE_CAT IS NULL
+        | ORDER BY TABLE_TYPE, TABLE_CAT, TABLE_SCHEM, TABLE_NAME
+        |""".stripMargin)
+
+    check(
+      """
+        | SELECT TABLE_CAT, TABLE_SCHEM, TABLE_NAME, TABLE_TYPE, REMARKS,
+        | TYPE_CAT, TYPE_SCHEM, TYPE_NAME,
+        | SELF_REFERENCING_COL_NAME, REF_GENERATION
+        | FROM system.jdbc.tables
+        | WHERE TABLE_CAT = 'ykf_catalog'
+        | ORDER BY TABLE_TYPE, TABLE_CAT, TABLE_SCHEM, TABLE_NAME
+        |""".stripMargin,
+      catalog = "ykf_catalog")
+
+    check(
+      """
+        | SELECT TABLE_CAT, TABLE_SCHEM, TABLE_NAME, TABLE_TYPE, REMARKS,
+        | TYPE_CAT, TYPE_SCHEM, TYPE_NAME,
+        | SELF_REFERENCING_COL_NAME, REF_GENERATION
+        | FROM system.jdbc.tables
+        | WHERE TABLE_CAT = 'ykf_catalog' AND TABLE_SCHEM IS NULL
+        | ORDER BY TABLE_TYPE, TABLE_CAT, TABLE_SCHEM, TABLE_NAME
+        |""".stripMargin,
+      catalog = "ykf_catalog")
+
+    check(
+      """
+        | SELECT TABLE_CAT, TABLE_SCHEM, TABLE_NAME, TABLE_TYPE, REMARKS,
+        | TYPE_CAT, TYPE_SCHEM, TYPE_NAME,
+        | SELF_REFERENCING_COL_NAME, REF_GENERATION
+        | FROM system.jdbc.tables
+        | WHERE TABLE_CAT = 'ykf_catalog' AND TABLE_SCHEM LIKE '%aa' ESCAPE '\'
+        | ORDER BY TABLE_TYPE, TABLE_CAT, TABLE_SCHEM, TABLE_NAME
+        |""".stripMargin,
+      catalog = "ykf_catalog",
+      "%aa")
+
+    check(
+      """
+        | SELECT TABLE_CAT, TABLE_SCHEM, TABLE_NAME, TABLE_TYPE, REMARKS,
+        | TYPE_CAT, TYPE_SCHEM, TYPE_NAME,
+        | SELF_REFERENCING_COL_NAME, REF_GENERATION
+        | FROM system.jdbc.tables
+        | WHERE TABLE_CAT = 'ykf_catalog' AND TABLE_NAME LIKE '%aa' ESCAPE '\'
+        | ORDER BY TABLE_TYPE, TABLE_CAT, TABLE_SCHEM, TABLE_NAME
+        |""".stripMargin,
+      catalog = "ykf_catalog",
+      tableName = "%aa")
+
+    check(
+      """
+        | SELECT TABLE_CAT, TABLE_SCHEM, TABLE_NAME, TABLE_TYPE, REMARKS,
+        | TYPE_CAT, TYPE_SCHEM, TYPE_NAME,
+        | SELF_REFERENCING_COL_NAME, REF_GENERATION
+        | FROM system.jdbc.tables
+        | WHERE TABLE_CAT = 'ykf_catalog' AND TABLE_NAME LIKE '%aa' ESCAPE '\' AND FALSE
+        | ORDER BY TABLE_TYPE, TABLE_CAT, TABLE_SCHEM, TABLE_NAME
+        |""".stripMargin,
+      catalog = "ykf_catalog",
+      tableName = "%aa",
+      emptyRes = true)
+
+    check(
+      """
+        | SELECT TABLE_CAT, TABLE_SCHEM, TABLE_NAME, TABLE_TYPE, REMARKS,
+        | TYPE_CAT, TYPE_SCHEM, TYPE_NAME,
+        | SELF_REFERENCING_COL_NAME, REF_GENERATION
+        | FROM system.jdbc.tables
+        | WHERE TABLE_CAT = 'ykf_catalog' AND TABLE_SCHEM LIKE '%aa' ESCAPE '\' AND
+        | TABLE_TYPE IN ('ykf_type', 'ykf2_type') ORDER BY TABLE_TYPE, TABLE_CAT, TABLE_SCHEM, TABLE_NAME
```

Comment:
It seems to be identified `'\' AND TABLE_TYPE IN ('` instead of `SEARCH_STRING_ESCAPE  + AND + tableTypeFilter `, that is a errorToken, do you have any idea? @ulysses-you

Reply:
seems `'\'` is strict, how about relaxing it to `stirngInValue` ?